src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -380,14 +377,16 @@ class AgoraBusiness(permissionsDataSource: PermissionsDataSource)(implicit ec: E
         if (configs.isEmpty) configs else {
           // from the method snapshot, get the WDL and parse it
           val wdl = methodSnapshot.payload.getOrElse(throw AgoraException(s"Method $namespace:$name:$snapshotId is missing a payload."))
-          parseWDL(wdl) match {
+
+          WaasClient.describe(wdl, accessToken) match {
             case Failure(ex) => throw ex
             case Success(workflow) =>
               // get the set of required input keys, optional input keys, and output keys for this WDL
-              val (optionalInputs, requiredInputs) = workflow.inputs.partition(_._2.optional)
-              val optionalInputKeys = optionalInputs.keySet.map(_.toString)
-              val requiredInputKeys = requiredInputs.keySet.map(_.toString)
-              val outputKeys:Set[String] = workflow.outputs.map(_.fullyQualifiedName.toString).toSet
+              // We expect fully qualified names, but WaaS doesn't return that. So we construct it here
+              val (optionalInputs, requiredInputs) = workflow.getInputs.asScala.partition(_.getOptional)
+              val optionalInputKeys = optionalInputs.map(workflow.getName + "." + _.getName).toSet
+              val requiredInputKeys = requiredInputs.map(workflow.getName + "." + _.getName).toSet
+              val outputKeys:Set[String] = workflow.getOutputs().asScala.map(workflow.getName + "." + _.getName).toSet
```

Comment:
Interesting approach -- I think it's 6 of one, half dozen of the other just like you point out.  I'm inclined to leave it since I think the existing one is a little more straightforward to understand